featuretools/entityset/entity.py

```diff
@@ -447,10 +448,7 @@ def set_index(self, variable_id, unique=True):
             self.df = self.df.set_index(self.df[variable_id], drop=False)
             self.df.index.name = None
             if unique:
-                if isinstance(self.df.index, dd.core.Index):
-                    index_is_unique = self.df[variable_id].compute().is_unique
-                else:
-                    index_is_unique = self.df[variable_id].is_unique
+                index_is_unique = self.df[variable_id].is_unique
                 assert index_is_unique, "Index is not unique on dataframe (Entity {})".format(self.id)
```

Comment:
I think we can eliminate the creation of the `index_is_unique` variable and update the assert to match `master`:

`assert self.df.index.is_unique, "Index is not unique on dataframe (Entity {})".format(self.id)`